backend/src/schema/resolvers/posts.js

```diff
@@ -77,14 +77,16 @@ export default {
   Mutation: {
     CreatePost: async (_parent, params, context, _resolveInfo) => {
       const { categoryIds } = params
+      const { image: imageInput } = params
       delete params.categoryIds
-      params = await fileUpload(params, { file: 'imageUpload', url: 'image' })
+      delete params.image
       params.id = params.id || uuid()
       const session = context.driver.session()
       const writeTxResultPromise = session.writeTransaction(async transaction => {
         const createPostTransactionResponse = await transaction.run(
           `
-            CREATE (post:Post {params})
+            CREATE (post:Post)
+            SET post += $params
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-20T19:52:23Z" title="Thursday, February 20th 2020, 8:52:23 pm +01:00">Feb 20, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -77,14 +77,16 @@ export default {
   Mutation: {
     CreatePost: async (_parent, params, context, _resolveInfo) => {
       const { categoryIds } = params
+      const { image: imageInput } = params
       delete params.categoryIds
-      params = await fileUpload(params, { file: 'imageUpload', url: 'image' })
+      delete params.image
       params.id = params.id || uuid()
       const session = context.driver.session()
       const writeTxResultPromise = session.writeTransaction(async transaction => {
         const createPostTransactionResponse = await transaction.run(
           `
-            CREATE (post:Post {params})
+            CREATE (post:Post)
+            SET post += $params
```


why should this be `+=` ?? 
it's creating a post, so it should never have any `params` to add to, oder?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-02-22T12:45:38Z" title="Saturday, February 22nd 2020, 1:45:38 pm +01:00">Feb 22, 2020</time>_

---


You're right. But it does not hurt either, or does it? :thinking: